CMakeLists.txt

```diff
@@ -42,6 +42,21 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Build Halide with ccache if the package is present
+set(Halide_CCACHE_BUILD OFF CACHE BOOL "Set to ON for a ccache enabled build")
+if (Halide_CCACHE_BUILD)
+  find_program(CCACHE_PROGRAM ccache)
+  if (CCACHE_PROGRAM)
+      set(Halide_CCACHE_PARAMS "CCACHE_CPP2=yes CCACHE_HASHDIR=yes"
+          CACHE STRING "Parameters to pass through to ccache")
+      set(CCACHE_PROGRAM "${Halide_CCACHE_PARAMS} ${CCACHE_PROGRAM}")
+      set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${CCACHE_PROGRAM})
```

Comment:
We shouldn't ever be enabling ccache on Windows, since it doesn't exist there. (But I'll make the change.)

Reply:
(Also note that the 'set-global-property' approach here is taken directly from LLVM)

Reply:
Just in general: LLVM's build is not exactly a model of good CMake code.

I know ccache isn't a thing on Windows _now_, but it might be in the future. There's an open PR about it. Granted, it doesn't look like it's going to get merged soon.

Reply:
There is a thing called `clcache` which intends to be a workalike. Haven't investigated.